### PR TITLE
Add AT TIME ZONE syntax

### PR DESCRIPTION
**Pull request by marregui**
## Checklist

 - [x] User relevant changes are recorded in ``CHANGES.txt``
 - [x] Touched code is covered by tests
 - [x] Documentation has been updated if necessary
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

File: sql-parser/src/main/antlr/SqlBase.g4 [hunk 1]
```diff
@@ -237,6 +237,7 @@ primaryExpression
     | value=primaryExpression '[' index=valueExpression ']'                          #subscript
     | ident ('.' ident)*                                                             #dereference
     | primaryExpression CAST_OPERATOR dataType                                       #doubleColonCast
+    | timestamp=primaryExpression AT TIME ZONE zone=stringLiteralOrNull              #atTimezone
```
Comment (kovrus) [inline sql-parser/src/main/antlr/SqlBase.g4]: Is `zone` supports only literals? In pg it can also be a column reference.

Comment (marregui) [reply sql-parser/src/main/antlr/SqlBase.g4]: My interpretation of https://www.postgresql.org/docs/current/functions-datetime.html#FUNCTIONS-DATETIME-ZONECONVERT where it says "In these expressions, the desired time zone zone can be specified either as a text string (e.g., 'America/Los_Angeles') or as an interval (e.g., INTERVAL '-08:00')." is that only literals and null are allowed.

Comment (marregui) [reply sql-parser/src/main/antlr/SqlBase.g4]: so no column names

Comment (mfussenegger) [reply sql-parser/src/main/antlr/SqlBase.g4]: Seems like postgres allows any expressions:

```
select current_timestamp AT TIME ZONE (select '+3:00');
          timezone
----------------------------
 2019-08-06 12:04:11.132376
```

Given that we re-write it to a function that already supports arbritray expressions I'd be in favour of lifting this restriction. I don't like adding yet another expression variant with `stringLiteralOrNull`

File: blackbox/docs/general/ddl/data-types.rst [hunk 2]
```diff
@@ -316,6 +316,70 @@ converted to UTC without considering the time zone indication.
...
+
+   "timestamp without time zone AT TIME ZONE zone", "timestamp with time zone", "Treat \
+   given time stamp without time zone as located in the specified time zone"
+   "timestamp with time zone AT TIME ZONE zone", "timestamp without time zone", "Convert \
+   given time stamp with time zone to the new time zone, with no time zone designation"
+
+In these expressions, the desired time zone is specified as a string
+(e.g., 'Europe/Madrid', '+02:00'). See :ref:`Timezone <date-format-timezone>`.
+
+Examples::
```
Comment (mfussenegger) [inline blackbox/docs/general/ddl/data-types.rst]: I think we could shorten the examples here a bit and instead refer to the existing scalar function, given that this is just syntax sugar.

File: sql-parser/src/main/antlr/SqlBase.g4 (hunk 1 above)
Comment (kovrus) [reply sql-parser/src/main/antlr/SqlBase.g4]: it is also not consistent with our `timezone` scalar, where zone can be an expression. so literal here is an artificial limitation.

File: sql/src/test/java/io/crate/expression/scalar/timestamp/AtTimezoneSyntaxFunctionTest.java [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+import io.crate.types.DataTypes;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import java.time.format.DateTimeParseException;
+
+import static java.lang.String.format;
+
+public class AtTimezoneSyntaxFunctionTest extends AbstractScalarFunctionsTest {
```
Comment (mfussenegger) [inline sql/src/test/java/io/crate/expression/scalar/timestamp/AtTimezoneSyntaxFunctionTest.java]: I think we could instead write a test for the parser that verifies that we generate a `FunctionCall` and then only have a single test here, to verify the integration between `AT TIME ZONE` and the  `timezone` scalar. No need to test all edge cases again.

File: sql-parser/src/main/antlr/SqlBase.g4 (hunk 1 above)
Comment (marregui) [reply sql-parser/src/main/antlr/SqlBase.g4]: ok, on it

File: blackbox/docs/general/ddl/data-types.rst (hunk 2 above)
Comment (marregui) [reply blackbox/docs/general/ddl/data-types.rst]: Perfect, on it.

File: sql/src/test/java/io/crate/expression/scalar/timestamp/AtTimezoneSyntaxFunctionTest.java (hunk 3 above)
Comment (marregui) [reply sql/src/test/java/io/crate/expression/scalar/timestamp/AtTimezoneSyntaxFunctionTest.java]: done

File: sql/src/test/java/io/crate/expression/scalar/timestamp/AtTimezoneSyntaxFunctionTest.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+import io.crate.sql.tree.FunctionCall;
+import io.crate.sql.tree.QualifiedName;
+import io.crate.sql.tree.StringLiteral;
+import org.junit.Test;
+import static org.junit.Assert.assertEquals;
+
+public class AtTimezoneSyntaxFunctionTest {
+
+    @Test
+    public void testCorrectParse() {
```
Comment (mfussenegger) [inline sql/src/test/java/io/crate/expression/scalar/timestamp/AtTimezoneSyntaxFunctionTest.java]:
```suggestion
    public void test_at_time_zone_is_parsed_as_timezone_function_call() {
```

**Comment by marregui** [comment]
retest this please
